lib/datasource.js

```diff
@@ -1464,7 +1464,7 @@ DataSource.prototype.discoverSchemas = function(modelName, options, cb) {
 
       schema.options.relations = {};
       foreignKeys.forEach(function(fk) {
-        var propName = nameMapper('column', fk.pkTableName);
+        var propName = nameMapper('fk', fk.fkName + 'Rel');
```

Comment:
This may be better:
```
var propName = nameMapper('fk', (fk.fkName || fk.pkTableName)) + 'Rel';
``` 

Or even move `+ 'Rel'` to the default name mapper implementation?